src/main/java/com/synopsys/integration/alert/web/certificates/CertificateActions.java

```diff
@@ -76,9 +76,15 @@ public CertificateModel createCertificate(CertificateModel certificateModel) thr
 
     private CertificateModel importCertificate(CertificateModel certificateModel) throws AlertException {
         CustomCertificateModel certificateToStore = convertToDatabaseModel(certificateModel);
-        CustomCertificateModel storedCertificate = certificateAccessor.storeCertificate(certificateToStore);
-        certificateUtility.importCertificate(storedCertificate);
-        return convertFromDatabaseModel(storedCertificate);
+        try {
+            CustomCertificateModel storedCertificate = certificateAccessor.storeCertificate(certificateToStore);
+            certificateUtility.importCertificate(storedCertificate);
+            return convertFromDatabaseModel(storedCertificate);
+        } catch (AlertException ex) {
+            certificateAccessor.deleteCertificate(certificateToStore.getAlias());
+            certificateUtility.removeCertificate(certificateToStore.getAlias());
```

Comment:
This line may also throw an exception and I'm not sure how we want to handle it.